comet/broker.py

```diff
@@ -408,22 +421,33 @@
         # wait for half of kotekans timeout before we admit we don't have it
         await lock_states.release()
         logger.debug("wait_for_state: Waiting for state {}".format(id))
+        wait_time = WAIT_TIME
+        start_wait = time.time()
         while True:
             # did someone send it to us by now?
             async with cond_states as r:
                 try:
-                    await asyncio.wait_for(cond_states.wait(), WAIT_TIME)
-                except (TimeoutError, CancelledError):
+                    await cond_states.wait(wait_time)
```

Comment:
I think this passes in a float still (even though it's rounded to the ceil int). You need to explicitly cast too I think.